vignettes/parallel.Rmd

```diff
@@ -0,0 +1,125 @@
+---
+title: "Running tests in parallel"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Running tests in parallel}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+library(testthat)
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+## Parallel testthat (preview)
+
+### Turning on parallel tests
+
+The current development version of testthat supports running test in
+parallel, in multiple R subprocesses. To turn on this feature, you
+need to set the `TESTTHAT_PARALLEL` environment variable to `true`.
+
+### Number of parallel subprocesses
+
+The number of subprocesses is taken from the `Ncpus` option, if set,
+otherwise it is fixed to 4. This will change later, see the TODO list
+at <https://github.com/r-lib/testthat/pull/1032>. (But testthat never
+starts more subprocesses than the number of test files.)
+
+### Test files, test order and state
+
+Testthat starts running the test files in alphabetical order. As soon
+as a subprocess has finished with a file, it receives another file to
+run from the main process, until all files are done. In general, the
+user cannot make any assumptions about the order of the files, and
+which subprocess they'll be executed on.
+
+Test files do not start in a clean R process currently. The first *n*
+files, where *n* is the number of subprocesses, do start in a clean R
+process, but users should not rely on that. In particular, options
+(set via `options()` or otherwise) are *not* reset, loaded packages
+are *not* unloaded, the global environment is *not* cleared, etc. In
+the future the user will be able to request a clean R process for a
+test file, see the TODO at
```

Comment:
Yeah, it is not a good default. But it would be convenient to be able to request it for some test files, that are special. E.g. you want to run your tests with a package (say tibble), but you also want to have a test file for the case where tibble is not loaded. For the latter you would need a clean session.

Btw. running everything in a clean session might be a good default for a CI, actually. Then even if setting up a session is a second, for 60 test files you only lose a minute.